rasa/utils/tensorflow/models.py

```diff
@@ -275,10 +278,26 @@ def rasa_predict(self, model_data: RasaModelData) -> Dict[Text, tf.Tensor]:
             )
 
         outputs = tf_utils.to_numpy_or_python_type(self._tf_predict_step(batch_in))
-        if np.size(outputs[DIAGNOSTIC_DATA]["attention_weights"]) == 0:
-            outputs[DIAGNOSTIC_DATA]["attention_weights"] = None
+        outputs[DIAGNOSTIC_DATA] = self._empty_lists_to_none_in_dict(
+            outputs[DIAGNOSTIC_DATA]
+        )
         return outputs
 
+    @staticmethod
+    def _empty_lists_to_none_in_dict(input_dict: Dict[Text, Any]) -> Dict[Text, Any]:
+        """Recursively replaces empty list or np array with None in a dictionary."""
+
+        def _recurse(x):
+            if not isinstance(x, dict):
+                if (isinstance(x, list) or isinstance(x, np.ndarray)) and np.size(
+                    x
+                ) == 0:
```

Comment:
can you just do `len(x)==0`?

Reply:
That gives an error if it's a scalar wrapped in a numpy array (e.g. `np.array(47)`).